web/src/stream_list.js

```diff
@@ -63,28 +68,33 @@ export function update_count_in_dom(
         // Normal stream, has unmuted unreads; display normally.
         ui_util.update_unread_count_in_dom($subscription_block, stream_counts.unmuted_count);
         $subscription_block.addClass("stream-with-count");
+        $subscription_block.removeClass("has-only-muted-unreads");
         $subscription_block.removeClass("has-unmuted-unreads");
     } else if (stream_counts.unmuted_count > 0 && stream_counts.stream_is_muted) {
         // Muted stream, has unmuted unreads.
         ui_util.update_unread_count_in_dom($subscription_block, stream_counts.unmuted_count);
         $subscription_block.addClass("stream-with-count");
+        $subscription_block.removeClass("has-only-muted-unreads");
         $subscription_block.addClass("has-unmuted-unreads");
     } else if (stream_counts.muted_count > 0 && stream_counts.stream_is_muted) {
         // Muted stream, only muted unreads.
         ui_util.update_unread_count_in_dom($subscription_block, stream_counts.muted_count);
         $subscription_block.addClass("stream-with-count");
+        $subscription_block.removeClass("has-only-muted-unreads");
         $subscription_block.removeClass("has-unmuted-unreads");
     } else if (stream_counts.muted_count > 0 && !stream_counts.stream_is_muted) {
         // Normal stream, only muted unreads: display nothing. The
         // current thinking is displaying those counts with muted
         // styling would be more distracting than helpful.
-        ui_util.update_unread_count_in_dom($subscription_block, 0);
+        ui_util.update_unread_count_in_dom($subscription_block, stream_counts.muted_count);
```

Comment:
I checked with @alya and we agree the plan was to only show an unread count for messages in muted topics on the stream row if there is at least one unread (muted) mention; so I think you need to do an if/else here that would make this 0 if there are no muted mentions.